### DISCUSSION TITLE
When should we clear bucket storage table if integrations with jdbc?

### DISCUSSION THREAD

**Discussion by VietKhoa2410** [vladimir-bukhtoyarov/bucket4j#340] [Q&A] [answered]
In [the document for JDBC integration](https://bucket4j.com/7.5.0/toc.html#jdbc-integrations), there are a principle that:
"You should create a trigger or a scheduler that will clear your bucket storage table since DBMS is not IMDB, and DBMS don’t give TTL the opportunity"
I wonder when and why should we clear that table? Is it wrong if I save one bucket  for each user and not clear it anymore?

**Comment by vladimir-bukhtoyarov** [comment, marked as answer]
It is supposed that you implement periodically cleaning the table based on last-modification time.

**Reply by alantonica** [reply]
Can you expand on this a little bit, pls? 
The "why" question still remains. 

If user A makes requests, i have a bucket for him. The bucket exists there and will refill its content according to what the configuration is and the user will consume its contents as they make requests. Why would I delete the bucket if I know the user still exists?
The only time, imo, we sohuld do that is when we actually delete the user A. 

Am i missing something? 🤔

**Reply by vladimir-bukhtoyarov** [reply]
It depends from nature of "user ID":
* If this is some random IP address from internet then it is better to remove it from bucket table as soon as it became inactive in order to preserve database volume. 
* If it is ID of some well known entity with clear life cycle and predictable amounts of these enitities(for example there are another tables where other peaces of information related to this entity is stored) then decision for TTL policy should be taken by yourself, including storing bucket for ever and removing only by FOREING KEY integrity chain.

**Comment by VietKhoa2410** [comment]
Thank you! I got it
